### PR TITLE
Add management API v1

### PR DESCRIPTION
**Pull request by yivo**

### CODE REVIEW

**Comment by antonverdi** [comment]
`spec/models/withdraw_spec.rb:216`
 you have after_commit callback on cancel event in withdraw model, which reverts created withdraw to canceled and tries to unlock funds. created event locks no funds. a possible fix is to ensure aasm_state in callback method `models/withdraw:152`: 
```
 def unlock_funds
    account.lock!
    account.unlock_funds sum, reason: Account::WITHDRAW_UNLOCK, ref: self unless aasm.from_state == :created
  end
```

**Comment by yivo** [comment]
image

File: app/api/management_api_v1/jwt_authentication_middleware.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    extend Memoist
+    mattr_accessor :security_configuration
+
+    def before
+      return if request.path == '/management_api/v1/swagger'
+      check_request_method!
+      check_query_parameters!
+      check_content_type!
+      payload = check_jwt!(jwt)
+      env['rack.input'] = StringIO.new(payload.fetch(:data, {}).to_json)
```
Comment (mod) [inline app/api/management_api_v1/jwt_authentication_middleware.rb:14]: Normally payload is base 64 encoded, please double check the RFC
```
   payload
      The "payload" member MUST be present and contain the value
      BASE64URL(JWS Payload).
```

**Review by mod — CHANGES_REQUESTED** [review]

File: app/api/management_api_v1/jwt_authentication_middleware.rb (hunk 1 above)
Comment (yivo) [reply app/api/management_api_v1/jwt_authentication_middleware.rb:14]: @mod It is handled by jwt-multisig gem. Check this https://github.com/rubykube/jwt-multisig/blob/master/lib/jwt-multisig.rb#L115

**Review by mod — APPROVED** [review]

**Pull request merged**